### DISCUSSION TITLE
Vuetify

### DISCUSSION THREAD

**Discussion by NinjaJake86** [tauri-apps/tauri#3026] [Q&A] [answered]
Hello!

I'm trying to get a Vuetify app bar to be a dragable titlebar but this doesn't seem to work when you add the data-tauri-drag-region tag to the element it is not draggable, anyone got any ideas?

Thanks! :)

**Comment by amrbashir** [comment]
What does your code look like ? Also open the dev tools and check if attribute is applied to the final HTML correctly.

**Comment by NinjaJake86** [comment, marked as answer]
So, for the app bar i have this (Minimal example)
```html
<template>
<v-app-bar data-tauri-drag-region flat dense app style="background: linear-gradient(to bottom, #9c1483, #c31432); user-select: none;">
</v-app-bar>
</template>
```

I've checked with inspect and the attribute does exist on the header element
```html
<header class="v-sheet theme--dark v-toolbar v-toolbar--dense v-toolbar--flat v-app-bar v-app-bar--fixed" data-booted="true" data-tauri-drag-region="" style="height: 48px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px; background: linear-gradient(rgb(156, 20, 131), rgb(195, 20, 50)); user-select: none;"></header>
```

**Reply by amrbashir** [reply]
Can you provide a minimal repro ? I can take a look and debug this faster

**Reply by NinjaJake86** [reply]
Here you go! :) Thanks for all the help, i do really appreciate it! https://github.com/NinjaJake86/Tauri-Vuetify-Drag-Example

**Reply by amrbashir** [reply]
It seems that `v-app-bar` renders two HTML elements, the outer one will have the attribute but the inner one, won't. When you click on a `v-app-bar`, the element that receives the click event is the inner one, and Tauri doesn't find that it has `data-tauri-drag-region` so it won't start dragging. you need to find a way to add it to the inner one.

**Reply by amrbashir** [reply]
We can definitely improve it a bit on our side, I will modify our detection to detect if the parent also has `data-tauri-drag-region`, thanks for catching this.

**Reply by NinjaJake86** [reply]
Thats great! Thank you for looking at this! Looking forward to the update. :)

**Reply by amrbashir** [reply]
Here is the PR https://github.com/tauri-apps/tauri/pull/3031 , if all goes well, the fix should be in the next release but it is better to find a way to put the tag directly on the element that receives the click.

**Reply by amrbashir** [reply]
A bit of an update: we can't do anything on our side without messing up other stuff. Seems like the best option for you is to find some other way to add the tag to the child.

In your Vuetify case, you can, in your mounted hook, get a reference to the `v-app-bar` element and apply the `data-tauri-drag-reigon` to its first child.

**Reply by NinjaJake86** [reply]
Okay, thanks for that! Following your advice i managed to get this working using the below code

`mounted() {
      let topBar = document.getElementById("topBar");
      topBar.children[0].setAttribute("data-tauri-drag-region", "");
    }`

Hope this helps anyone else who is experiencing the same issue! And thanks for all the help @amrbashir

**Reply by FabianLars** [reply]
@NinjaJake86 Hmm, you can probably just make the spacer actually fill the space and add the attribute to that one 🤔
